patroni/postgresql.py

```diff
@@ -990,13 +888,8 @@ def stop(self, mode='fast', block_callbacks=False, checkpoint=True, on_safepoint
         return success
 
     def _do_stop(self, mode, block_callbacks, checkpoint, on_safepoint):
-        if not self.is_running():
-            if self.data_directory_empty() and self._postmaster_cached_info['pid']:
-                pid = self.get_pid_with_lost_data_dir()
-                if pid > 0:
-                    self.terminate_starting_postmaster(pid)
-                    self.clean_postmaster_cached_info()
-                    return True, True
+        postmaster = self.is_running()
+        if not postmaster:
```

Comment:
Due to the check on https://github.com/zalando/patroni/pull/561/files#diff-d8c1582337c8426ef750420dc6b76949R707 it will not catch the case when data dir has been removed.

Also there should be mode='immediate': https://github.com/zalando/patroni/pull/561/files#diff-57c53e988b775ee2902488fb33c1f433R1028